mne/decoding/csp.py

```diff
@@ -129,8 +129,12 @@ def _fit(self, cov_a, cov_b):
         w_b = np.dot(np.dot(p, cov_b), p.T)
         # and solve it
         vals, vecs = linalg.eigh(w_a, w_b)
-        # sort vectors by discriminative power using eigen values
-        ind = np.argsort(np.maximum(vals, 1. / vals))[::-1]
+        # sort vectors by discriminative power using eigenvalues
+        ind = np.argsort(vals)[::-1]
+        vecs = vecs[:, ind]
+        # re-order (first, last, second, second last, third, ...)
+        ind[::2] = np.arange(0, np.ceil(len(ind)/2))
+        ind[1::2] = np.arange(len(ind) - 1, int(len(ind)/2) - 1, -1)
```

Comment:
also maybe you can avoid taking 3 times len(ind) and introduce a variable n_vals ?